doc/configuration.rst

```diff
@@ -211,6 +214,25 @@ consult the `regular expressions`_ module for more details.
 
         $ sphinx-build -D sphinx_gallery_conf.filename_pattern=plot_specific_example\.py ...
 
+To parse examples in other languages, add their extensions to ``example_extensions``.
+For example, to include examples in Python, Julia, and C++::
+
+    sphinx_gallery_conf = {
+        ...
+        'example_extensions': {'.py', '.jl', '.cpp'}
+    }
+
+Parsing and syntax highlighting is supported by the Pygments library, with the language
+determined by the file extension. To override Pygments' default file associations, the
+``filetype_parsers`` option can be used to specify a ``dict`` mapping a file extension
+to any of the `pygments language names <https://pygments.org/languages/>`__. For
+example::
```

Comment:
I tried to clarify. This option isn't so much about handling the case of an extension with no corresponding lexer (though it can be used that way), but for the case where an extension corresponds to several lexers and Pygments' defaults to the wrong one. For example, Pygments lists `'.m` as a known extension for Matlab, Objective-C, and Mason, with the default being Objective-C.